include/lsst/afw/formatters/PropertyListFormatter.h

```diff
@@ -0,0 +1,91 @@
+// -*- lsst-c++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2008, 2009, 2010 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_FORMATTERS_DECORATEDIMAGEFORMATTER_H
+#define LSST_AFW_FORMATTERS_DECORATEDIMAGEFORMATTER_H
+
+/** @file
+ * @brief Interface for PropertyListFormatter class
+ *
+ * @author $Author: ktlim $
+ * @version $Revision: 2377 $
+ * @date $Date$
+ *
+ * Contact: Kian-Tat Lim (ktl@slac.stanford.edu)
+ * @ingroup afw
+ */
+
+/** @class lsst::afw::formatters::PropertyListFormatter
+ * @brief Class implementing persistence and retrieval for PropertyLists.
+ *
+ * @ingroup afw
+ */
+
+#include "lsst/daf/base.h"
+#include "lsst/daf/persistence.h"
+#include "lsst/pex/policy/Policy.h"
+
+namespace lsst {
+namespace afw {
+namespace formatters {
+
+class PropertyListFormatter : public lsst::daf::persistence::Formatter {
+public:
+    virtual ~PropertyListFormatter(void);
+
+    virtual void write(
+        lsst::daf::base::Persistable const* persistable,
+        lsst::daf::persistence::Storage::Ptr storage,
+        lsst::daf::base::PropertySet::Ptr additionalData
+    );
+    virtual lsst::daf::base::Persistable* read(
+        lsst::daf::persistence::Storage::Ptr storage,
+        lsst::daf::base::PropertySet::Ptr additionalData
+    );
+    virtual void update(
+        lsst::daf::base::Persistable* persistable,
+        lsst::daf::persistence::Storage::Ptr storage,
+        lsst::daf::base::PropertySet::Ptr additionalData
+    );
+
+    static lsst::daf::persistence::Formatter::Ptr createInstance(
+        lsst::pex::policy::Policy::Ptr policy
```

Comment:
I changed all these to PTR() and had a template match error (that was actually caused by changing `PropertySet` to `PropertyList`).  I'll go through and change them all to `std::shared_ptr<T>`.